src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Infer.java

```diff
@@ -651,6 +654,90 @@ public Type instantiateFunctionalInterface(DiagnosticPosition pos, Type funcInte
             return owntype;
         }
     }
+
```

Comment:
nit: before we push this I think some javadoc, not to formal, should be added explaining what this method is doing etc, now we remember but later on we could forget some details, also a reference to the section of the spec will be very helpful